practical-react-unit-testing-with-jest-mocks-and-spies/Exercise1/solution/src/Select.test.tsx

```diff
@@ -0,0 +1,51 @@
+/**
+ * @jest-environment jsdom
+ */
+//manually mocking toSelectOptions
+
+import { render } from "@testing-library/react";
+import { Select } from "./Select";
+import { toSelectOptions } from "./toSelectOptions";
+
+jest.mock("./toSelectOptions", () => ({
+  toSelectOptions: jest.fn().mockReturnValue([
+    { label: "2021", value: "2021" },
+    { label: "2022", value: "2022" },
+    { label: "2023", value: "2023" },
+  ]),
+}));
+describe("Year", () => {
+  test("Year works with one-off mocked values", () => {
+    toSelectOptions.mockImplementationOnce(() => {
+      return [{ label: "2024", value: "2024" }]
+    })
+
+    const result = render(
+      <Select
+        label="Year"
+        name="year"
+        value="2023"
+        options={toSelectOptions(["2020"])}
+      />
+    );
+    const selector = result.getByText("Select...");
+    expect(selector).toBeDefined()
+    expect(selector.querySelector("option[value='2024']")).toBeDefined()
```

Comment:
Another good suggestion:

> Avoid direct Node access. Prefer using the methods from Testing Library.eslint(testing-library/no-node-access)

Reply:
Is there a method in Testing Library that will find an option?  I was having trouble accessing it directly with getByLabel()

Reply:
Answering my own question:
- getByText() does return something...
- but the something is the option
- so just use `screen.getByText()` to get each option.